### PR TITLE
Developer Guide: Update Design (Architecture, Logic), Use Cases (List Appointments) Section

### LINKED ISSUE #23: Developer Guide: Design (Architecture, Logic), Use Cases (List Appointments)

### PR DESCRIPTION
**Pull request by glatiuden**
- Add introduction
- Update design - architecture & logic sections
- Update user stories
- Add use case for `list_appointments`

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -250,44 +263,66 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | add a new person               |                                                                        |
-| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
-
+| -------- | --------------------------------------------- | --------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- |
```
Comment (kingsleykuan) [inline docs/DeveloperGuide.md:264]: Sorry I think there was some miscommunication as I also updated the user stories! When merging you can see if you want to keep your version or mine, your version looks good as well.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -250,44 +263,66 @@ _{Explain here how the data archiving feature will be implemented}_
...
-**MSS**
-
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
-
+### Appointment
+<pre>
+Use Case UC001: Add an appointment
```
Comment (kingsleykuan) [inline docs/DeveloperGuide.md]: Is this use case not completed yet?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -250,44 +263,66 @@ _{Explain here how the data archiving feature will be implemented}_
...
+<pre>
+Use Case UC003: List tuition appointments
+View the list of personal tuition appointments.
 
-* 3a. The given index is invalid.
+<b>System: Tutor Tracker</b>
+<b>Actor: User</b>
 
-    * 3a1. AddressBook shows an error message.
+<b>Preconditions: Appointment list must exist.</b>
```
Comment (kingsleykuan) [inline docs/DeveloperGuide.md]: Is this precondition required? I think it is fair to assume that a list always exists and it might just be empty.

**Review by kingsleykuan — CHANGES_REQUESTED** [review]
Some of the changes in this pull request are duplicates of #2 and need to be removed, otherwise there will be many merge conflicts when trying to merge both of them.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (glatiuden) [reply docs/DeveloperGuide.md]: This meant to serve as a base template as I modified the styling slightly. Can choose to adopt this styling or go with the default style from AB3.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (glatiuden) [reply docs/DeveloperGuide.md]: Noted, I have removed it & committed.

**Review by kingsleykuan — DISMISSED** [review]

**Comment by glatiuden** [comment]
Will resolve the conflicts after PR #2 is approved due to accidental merging of branches.

**Review by jared98lyj — DISMISSED** [review]

**Review by kingsleykuan — DISMISSED** [review]

**Review by jared98lyj — APPROVED** [review]

**Review by kingsleykuan — APPROVED** [review]

**Pull request merged**